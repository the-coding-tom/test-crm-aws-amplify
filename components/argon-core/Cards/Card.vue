<template>
  <div
    :class="[
      {'card-lift--hover': hover},
      {'shadow': shadow},
      {[`shadow-${shadowSize}`]: shadowSize},
      {[`bg-gradient-${gradient}`]: gradient},
      {[`bg-${type}`]: type}
    ]"
    class="card">

    <slot name="image"/>
    <div
      v-if="$slots.header"
      :class="headerClasses"
      class="card-header">
      <slot name="header"/>
    </div>
    <div
      v-if="!noBody"
      :class="bodyClasses"
      class="card-body">
      <slot/>
    </div>

    <slot v-if="noBody"/>

    <div
      v-if="$slots.footer"
      :class="footerClasses"
      class="card-footer">
      <slot name="footer"/>
    </div>
  </div>
</template>
<script>
export default {
  name: 'Card',
  props: {
    type: {
      type: String,
      description: 'Card type',
      default: null
    },
    gradient: {
      type: String,
      description: 'Card background gradient type (warning,danger etc)',
      default: null
    },
    hover: {
      type: Boolean,
      description: 'Whether card should move on hover',
      default: null
    },
    shadow: {
      type: Boolean,
      description: 'Whether card has shadow',
      default: null
    },
    shadowSize: {
      type: String,
      description: 'Card shadow size',
      default: null
    },
    noBody: {
      type: Boolean,
      default: false,
      description: 'Whether card should have wrapper body class'
    },
    bodyClasses: {
      type: [String, Object, Array],
      description: 'Card body css classes',
      default: null
    },
    headerClasses: {
      type: [String, Object, Array],
      description: 'Card header css classes',
      default: null
    },
    footerClasses: {
      type: [String, Object, Array],
      description: 'Card footer css classes',
      default: null
    }
  }
}
</script>
<style>
</style>
