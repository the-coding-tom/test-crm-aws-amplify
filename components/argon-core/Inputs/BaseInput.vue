<template>
  <div class="form-group">
    <slot name="label">
      <label
        v-if="label"
        :class="labelClasses">
        {{ label }}
      </label>
    </slot>
    <div
      :class="[
        {'input-group': hasIcon},
        {'focused': focused},
        {'input-group-alternative': alternative},
        {'has-label': label || $slots.label},
        inputGroupClasses
    ]">
      <div
        v-if="prependIcon || $slots.prepend"
        class="input-group-prepend">
        <span class="input-group-text">
          <slot name="prepend">
            <i :class="prependIcon"/>
          </slot>
        </span>
      </div>
      <slot v-bind="slotData">
        <input
          :value="value"
          :type="type"
          v-bind="$attrs"
          :valid="!error"
          :required="required"
          :class="[{'is-valid': valid === true}, {'is-invalid': error}, inputClasses]"
          class="form-control"
          v-on="listeners">
      </slot>
      <div
        v-if="appendIcon || $slots.append"
        class="input-group-append">
        <span class="input-group-text">
          <slot name="append">
            <i :class="appendIcon"/>
          </slot>
        </span>
      </div>
      <slot name="infoBlock"/>
      <slot name="error">
        <div
          v-if="error"
          class="invalid-feedback"
          style="display: block;">
          {{ error }}
        </div>
      </slot>
      <slot name="success">
        <div
          v-if="!error && valid"
          class="valid-feedback">
          {{ successMessage }}
        </div>
      </slot>
    </div>
  </div>
</template>
<script>
export default {
  name: 'BaseInput',
  inheritAttrs: false,
  props: {
    required: {
      type: Boolean,
      description: 'Whether input is required (adds an asterix *)',
      default: true
    },
    group: {
      type: Boolean,
      description:
        'Whether input is an input group (manual override in special cases)',
      default: null
    },
    valid: {
      type: Boolean,
      description: 'Whether is valid',
      default: undefined
    },
    alternative: {
      type: Boolean,
      description: 'Whether input is of alternative layout',
      default: false
    },
    label: {
      type: String,
      description: 'Input label (text before input)',
      default: null
    },
    error: {
      type: String,
      description: 'Input error (below input)',
      default: null
    },
    successMessage: {
      type: String,
      description: 'Input success message',
      default: 'Looks good!'
    },
    labelClasses: {
      type: String,
      description: 'Input label css classes',
      default: 'form-control-label'
    },
    inputClasses: {
      type: String,
      description: 'Input css classes',
      default: null
    },
    inputGroupClasses: {
      type: String,
      description: 'Input group css classes',
      default: null
    },
    value: {
      type: [String, Number],
      description: 'Input value',
      default: null
    },
    type: {
      type: String,
      description: 'Input type',
      default: 'text'
    },
    appendIcon: {
      type: String,
      description: 'Append icon (right)',
      default: null
    },
    prependIcon: {
      type: String,
      description: 'Prepend icon (left)',
      default: null
    }
  },
  data() {
    return {
      focused: false
    }
  },
  computed: {
    listeners() {
      return {
        ...this.$listeners,
        input: this.updateValue,
        focus: this.onFocus,
        blur: this.onBlur
      }
    },
    slotData() {
      return {
        focused: this.focused,
        error: this.error,
        ...this.listeners
      }
    },
    hasIcon() {
      const { append, prepend } = this.$slots
      return (
        append !== undefined ||
        prepend !== undefined ||
        this.appendIcon !== undefined ||
        this.prependIcon !== undefined ||
        this.group
      )
    }
  },
  methods: {
    updateValue(evt) {
      let value = evt.target.value
      this.$emit('input', value)
    },
    onFocus(evt) {
      this.focused = true
      this.$emit('focus', evt)
    },
    onBlur(evt) {
      this.focused = false
      this.$emit('blur', evt)
    }
  }
}
</script>
<style>
</style>
