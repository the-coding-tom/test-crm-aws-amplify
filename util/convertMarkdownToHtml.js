export const convertMarkdownToHtml = text => {
  var showdown = require('showdown')
  var converter = new showdown.Converter()
  const eventDescription = text.replace(/([ ]*\n)/g, `<br/>\n`)
  const finalDescription = eventDescription.replace(
    /(<br\/>\n<br\/>\n<br\/>)/g,
    `<br/><br/>`
  )
  return converter.makeHtml(finalDescription)
}
