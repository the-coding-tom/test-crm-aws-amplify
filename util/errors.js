export const displayError = function(e) {
  if (e.response) {
    processError(e.response.data.message)

    if (e.response.data.errors) {
      _.each(e.response.data.errors, (o, i) => {
        return _.each(o, (v, k) => processError(v))
      })
    }

    return
  }

  processError(e.message)
}

const processError = function(message) {
  console.log('------------------------------------')
  console.log(message)
  console.log('------------------------------------')
}
