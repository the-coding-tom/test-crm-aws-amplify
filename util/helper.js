const randomColor = () => {
  let letters = 'ABCDEF0123456789'
  let len = 6
  let result = '#'
  for (let i = 0; i < len; i++) {
    result += letters[parseInt(Math.random() * letters.length)]
  }
  return result
}

const notify = {
  error: {
    title: 'Error',
    variant: 'danger',
    solid: true
  },
  sucess: {
    title: 'Success',
    variant: 'success',
    solid: true
  }
}

const parseRoomdate = (data, vm) => {
  const roomAvaila = data.split(',')
  const roomAvailability = []
  roomAvaila.forEach(data => {
    let weekday = []
    let timeof = []
    let lookup = {
      su: 0,
      mo: 1,
      tu: 2,
      we: 3,
      th: 4,
      fr: 5,
      sa: 6
    }
    const timeData = data.split(' ')
    if (timeData.length === 2) {
      weekday.push(timeData[0])
      timeof.push(timeData[1])
    }
    if (timeData.length === 3) {
      weekday.push(timeData[1])
      timeof.push(timeData[2])
    }
    const splitWeekday = weekday[0].split('-')
    const splitTimeOf = timeof[0].split('-')
    const AStruc = {
      from: '',
      to: '',
      weekdays: []
    }
    splitWeekday.forEach(week => {
      AStruc.weekdays.push(lookup[week])
    })
    AStruc.from = vm.$moment(splitTimeOf[0], 'HH').format('hh:mm')
    AStruc.to = vm.$moment(splitTimeOf[1], 'HH').format('hh:mm')
    roomAvailability.push(AStruc)
  })
  return roomAvailability
}

const undoParse = (data, vm) => {
  let str = ''
  const lookup = {
    0: 'su',
    1: 'mo',
    2: 'tu',
    3: 'we',
    4: 'th',
    5: 'fr',
    6: 'sa'
  }
  data.forEach(item => {
    let available = ''
    item.weekdays.forEach(day => {
      available += `${lookup[day]}-`
    })
    available = available.slice(0, -1)
    let weekdayfrom = item.from.charAt(0) + item.from.charAt(1)
    let weekdayto = item.to.charAt(0) + item.to.charAt(1)
    available += ` ${weekdayfrom}-${weekdayto}`
    str += `${available},`
  })
  str = str.slice(0, -1)
  return str
}

export default { randomColor, parseRoomdate, notify, parseRoomdate, undoParse }
