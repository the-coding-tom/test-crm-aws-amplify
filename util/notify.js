const notify = {
  error: {
    title: 'Error',
    variant: 'danger',
    solid: true
  },
  sucess: {
    title: 'Success',
    variant: 'success',
    solid: true
  }
}

export default notify
