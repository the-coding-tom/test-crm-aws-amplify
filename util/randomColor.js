const randomColor = () => {
  let letters = 'ABCDEF0123456789'
  let len = 6
  let result = '#'
  for (let i = 0; i < len; i++) {
    result += letters[parseInt(Math.random() * letters.length)]
  }
  return result
}

export default randomColor
