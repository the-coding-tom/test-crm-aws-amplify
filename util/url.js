export const getQueryParams = url => {
  const urlParam = require('url')

  const parsedUrl = urlParam.parse(url)

  return parsedUrl.search
}
