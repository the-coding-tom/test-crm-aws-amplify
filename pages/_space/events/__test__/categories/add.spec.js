import AddPage from '@/pages/_space/events/categories/add.vue'
import Card from '~/components/argon-core/Cards/Card.vue'
import BaseInput from '~/components/argon-core/Inputs/BaseInput.vue'
import BootstrapVue from 'bootstrap-vue'

import VueRouter from 'vue-router'
import cloneDeep from 'lodash/cloneDeep'
import flushPromises from 'flush-promises'
import { createWrapper } from '@/test.util.js'

import allCategories from '../resources/all-categories.json'
import newCategory from '../resources/new-category.json'

describe('Page: /:space/events/categories/add | Events Categories Add Page', () => {
  let NuxtStore, store, wrapper, $event, router

  beforeAll(async () => {
    NuxtStore = await import(`${process.env.buildDir}/store.js`)
    const { createRouter } = await import(`${process.env.buildDir}/router.js`)

    createRouter()

    $event = {
      getEventCategories: jest.fn().mockResolvedValue(cloneDeep(allCategories)),
      addCategory: jest.fn().mockResolvedValue()
    }
  })

  beforeEach(async () => {
    store = await NuxtStore.createStore()

    router = new VueRouter({
      routes: []
    })

    wrapper = createWrapper(
      AddPage,
      store,
      router,
      {
        name: null,
        color: null,
        loading: false
      },
      {},
      vueInstance => {
        vueInstance.use(BootstrapVue)
        vueInstance.component(BaseInput.name, BaseInput)
        vueInstance.component(Card.name, Card)
        vueInstance.prototype.$event = $event
      }
    )

    await wrapper.vm.$options.asyncData({ store, $event })
  })

  it('renders properly', async () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('adds a category', async () => {
    wrapper.find('#name').setValue(newCategory.name)
    wrapper.find('#color').setValue(newCategory.color)
    await wrapper.vm.$nextTick()

    wrapper.find('form').trigger('submit.prevent')
    await flushPromises()

    const categoryOptions = wrapper.findAll('tr td')
    expect(categoryOptions.length).toBe(6)
    expect(categoryOptions.at(0).text()).toBe(allCategories.data[0].name)
    expect(categoryOptions.at(1).text()).toBe(allCategories.data[0].color)
    expect(categoryOptions.at(2).text()).toBe(allCategories.data[1].name)
    expect(categoryOptions.at(3).text()).toBe(allCategories.data[1].color)
    expect(categoryOptions.at(4).text()).toBe(newCategory.name)
    expect(categoryOptions.at(5).text()).toBe(newCategory.color)
  })
})
