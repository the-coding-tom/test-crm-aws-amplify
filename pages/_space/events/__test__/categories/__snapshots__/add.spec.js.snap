// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: /:space/events/categories/add | Events Categories Add Page renders properly 1`] = `
<div>
  <form class="">
    <div class="header pb-6">
      <div class="container-fluid">
        <div class="header-body">
          <div class="d-flex justify-content-between py-4">
            <h1 class="page-title"><span>Add New Categories</span> <b style="display: none;">â€¢</b> </h1>
          </div>
        </div>
      </div>
    </div>
    <div class="container-fluid mt--6">
      <div class="card-deck flex-column flex-xl-row">
        <div class="card">
          <!---->
          <div class="card-body">
            <h3 class="mr-b-20">New Category</h3>
            <div class="row">
              <div class="form-group col-md-3"><label class="form-control-label">
                  Name
                </label>
                <div class="input-group has-label">
                  <!----> <input type="text" valid="true" required="required" id="name" placeholder="Name of Category" class="form-control">
                  <!---->
                  <!---->
                  <!---->
                </div>
              </div>
              <div class="form-group col-md-3"><label class="form-control-label">
                  Color Tag
                </label>
                <div class="input-group has-label">
                  <!----> <input type="text" valid="true" required="required" id="color" placeholder="#4C4D4F" class="form-control">
                  <!---->
                  <!---->
                  <!---->
                </div>
              </div>
              <div class="form-group col-md-3"><label>&nbsp;</label> <br> <button outline="" type="submit" class="btn sh-dashbtn btn-secondary">Add This Category</button></div>
            </div>
            <div class="mr-tb-80">
              <div class="m-n25">
                <table class="table table-hover table-striped">
                  <tbody>
                    <tr>
                      <td><i class="fa fa-circle" style="color: rgb(131, 172, 12);"></i>
                        talks
                      </td>
                      <td>#83ac0c</td>
                    </tr>
                    <tr>
                      <td><i class="fa fa-circle" style="color: rgb(202, 54, 139);"></i>
                        meetups
                      </td>
                      <td>#ca368b</td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
          <!---->
          <!---->
        </div>
      </div>
    </div>
  </form>
</div>
`;
